### DISCUSSION TITLE
Error while running etcd v3 on Docker

### DISCUSSION THREAD

**Discussion by lijie123bes** [etcd-io/etcd#18537] [Q&A] [answered]
### Bug report criteria

- [ ] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [ ] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [ ] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

I have an error:

docker: Error response from daemon: invalid header field value "oci runtime error: container_linux.go:247: starting container process caused \"exec: \\\"-name\\\": executable file not found in $PATH\"\n".

### What did you expect to happen?

normal start

### How can we reproduce it (as minimally and precisely as possible)?

When I try to run etcd (version 3.0.0) on Docker:

sudo docker run -d -v /usr/share/ca-certificates/:/etc/ssl/certs -p 4001:4001 -p 2380:2380 -p 2379:2379 \
 --name etcd quay.io/coreos/etcd:v3.0.0  \
 -name etcd0 \
 -advertise-client-urls http://${HostIP}:2379,http://${HostIP}:4001 \
 -listen-client-urls http://0.0.0.0:2379,http://0.0.0.0:4001 \
 -initial-advertise-peer-urls http://${HostIP}:2380 \
 -listen-peer-urls http://0.0.0.0:2380 \
 -initial-cluster-token etcd-cluster-1 \
 -initial-cluster etcd0=http://${HostIP}:2380 \
 -initial-cluster-state new

### Etcd version (please run commands below)

version 3.0.0

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by ivanvc** [comment, marked as answer]
Hi @lijie123bes, Thanks for opening this issue. This request is more of a support question than a bug, so I'll transfer it to a discussion.

Trying to solve the issue:
1. First, I noticed you're trying to use etcd v3.0.0. The oldest supported minor version is v3.4; I suggest using the latest version from that minor, v3.4.33 (or v3.5.15).
2. The docker image doesn't define an entry point, so if you pass arguments, you must specify the command first.

So, this should work (I verified it does, indentation for clarity):

```sh
docker run -d -v /usr/share/ca-certificates/:/etc/ssl/certs \
           -p 4001:4001 -p 2380:2380 -p 2379:2379 \
           --name etcd \
           quay.io/coreos/etcd:v3.4.33 \
               etcd \
                   -name etcd0 \
                   -advertise-client-urls http://${HostIP}:2379,http://${HostIP}:4001 \
                   -listen-client-urls http://0.0.0.0:2379,http://0.0.0.0:4001 \
                   -initial-advertise-peer-urls http://${HostIP}:2380 \
                   -listen-peer-urls http://0.0.0.0:2380/ \
                   -initial-cluster-token etcd-cluster-1 \
                   -initial-cluster etcd0=http://${HostIP}:2380 \
                   -initial-cluster-state new
```
